dramatiq/cli.py

```diff
@@ -385,7 +399,10 @@ def stop_worker_processes(signum):
 
         for proc in worker_processes:
             try:
-                os.kill(proc.pid, signum)
+                if proc.is_alive():
+                    os.kill(proc.pid, signum)
+                    # For Windows, give the process time to clean up
```

Comment:
Could we potentially use `os.waitpid` here?

Reply:
It appears that this issue is related to timing and how Windows handles the kill signal. The child process sometimes terminates between the check (``proc.is_alive()``) and the kill. If the main process sends the kill signal, Windows assigns the signal (SIGTERM is 15 in this case) to the child process return code, so the main process return code gets set to 15.

Depending upon the process timing, the child may have terminated before ``proc.is_alive()``, so we never send a kill signal and don't unintentionally modify the return code--the happy path. Occasionally, the process hasn't terminated and gets killed, causing the intermittent failures.

Using a one-second timeout on a join call is cleaner and achieves the same result.